app/controllers/users_controller.rb

```diff
@@ -213,9 +213,19 @@ def change_language(user_data)
   def change_email(user_data)
     @user.unconfirmed_email = user_data[:email]
     if @user.save
-      @user.send_confirm_email
-      if @user.unconfirmed_email
-        flash.now[:notice] = t("users.update.unconfirmed_email_changed")
+      if AppConfig.mail.enable?
+        @user.send_confirm_email
+        if @user.unconfirmed_email
+          flash.now[:notice] = t("users.update.unconfirmed_email_changed")
+        end
+      else
+        @user.email = @user.unconfirmed_email
```

Comment:
The `email` has also a unique index on it, because we need to validate duplicates when somebody registers with an already used mail-address. If there are other checks only on `unconfirmed_email`, we should probably add them to `email` too, because `unconfirmed_email` is only to store the new mail-address temporarily until it is confirmed, and we don't need that when we cant confirm it.